client/state/connectionMiddleware/reducer.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	OFFLINE_QUEUE_ADD,
+	OFFLINE_QUEUE_REMOVE
+} from 'state/action-types';
+
+export function actionQueue( state = [], action ) {
+	if ( action.type === OFFLINE_QUEUE_ADD ) {
+		let shouldQueueAction = true;
+
+		if ( action.squash ) {
+			const previousCalls = state.filter( storedAction =>
+				( storedAction.hash === action.hash )
+			);
+			shouldQueueAction = ( previousCalls.length === 0 );
+		}
```

Comment:
Another thing for fun. I'm simply writing down some thoughts as I review to make it easier for me to know what's going on.

```js
import { matchesProperty } from 'lodash';

const { squash: shouldSquash, hash } = action;

const shouldQueueAction = ! ( shouldSquash && state.some( matchesProperty( 'hash', hash ) ) );
```